src/main/java/frc/robot/subsystems/lifter/LifterStateHandler.java

```diff
@@ -0,0 +1,23 @@
+package frc.robot.subsystems.lifter;
+
+import edu.wpi.first.wpilibj2.command.Command;
+
+public class LifterStateHandler {
+
+	private final Lifter lifter;
+
+	public LifterStateHandler(Lifter lifter) {
+		this.lifter = lifter;
+	}
+
+	public Command setState(LifterState state) {
+		return switch (state) {
+			case HOLD -> lifter.getCommandsBuilder().stop();
+			case FORWARD -> lifter.getCommandsBuilder().setPower(state.getPower());
+			case BACKWARD -> lifter.getCommandsBuilder().setPower(state.getPower());
+			case EXTENDED -> lifter.getCommandsBuilder().setPower(state.getPower()).until(() -> lifter.isHigher(state.getTargetPosition()));
+			case RETRACTED -> lifter.getCommandsBuilder().setPower(state.getPower()).until(() -> lifter.isLower(state.getTargetPosition()));
```

Comment:
do the until in the climb structure, the states are never ending states

Reply:
no, they are not an infinite state. they have a end exactly like PID have an "end". because they do not have PID I think its ok o put the .until here because its bonded with the retracted funcition